src/unix/lwt_unix.cppo.ml

```diff
@@ -1149,6 +1149,14 @@ let chdir name =
   else
     run_job (Jobs.chdir_job name)
 
+external getcwd_job : unit -> string job = "lwt_unix_getcwd_job"
+
+let getcwd () =
+  if Sys.win32 then
+    Lwt.return (Unix.getcwd ())
```

Comment:
Yep, this is what is meant by realtively poor Windows support :/ A lot of it will get better when #328 is resolved. However, libuv doesn't include a non-blocking `getcwd`, so we will still need to have it threaded, and address Windows. I believe uwt does that to a great extent. The first thing to do for that would be to check whether the underlying Windows function can block or not – if it can't, then we don't have anything to worry about.

IMO, it's okay to put this off until later.